src/dist/replication/lib/replica_config.cpp

```diff
@@ -410,19 +410,29 @@ void replica::on_update_configuration_on_meta_server_reply(error_code err, dsn_m
 
         if (err != ERR_INVALID_VERSION)
         {
-            rpc_address target(_stub->_failure_detector->get_servers());
-            _primary_states.reconfiguration_task = rpc::create_rpc_response_task(
-                request,
+            //when the rpc call timeout, we would delay to do the recall
+            dsn_msg_add_ref(request); // will be released after recall
+            tasking::enqueue(
```

Comment:
sure, fixed

On Thu, Jul 28, 2016 at 3:40 PM, Weijie Sun <notifications@github.com>
wrote:

> In src/dist/replication/lib/replica_config.cpp
> <https://github.com/imzhenyu/rDSN/pull/497#discussion_r72576940>:
>
> > @@ -410,19 +410,29 @@ void replica::on_update_configuration_on_meta_server_reply(error_code err, dsn_m
> >
> >          if (err != ERR_INVALID_VERSION)
> >          {
> > -            rpc_address target(_stub->_failure_detector->get_servers());
> > -            _primary_states.reconfiguration_task = rpc::create_rpc_response_task(
> > -                request,
> > +            //when the rpc call timeout, we would delay to do the recall
> > +            dsn_msg_add_ref(request); // will be released after recall
> > +            tasking::enqueue(
>
> should store the return value in case the task is cancelled.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/497/files/d8635e3fc4fb2e12960d1e592fce66a24bfaedea#r72576940>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgQPXAPbiaZELWZvubfV31vWisHBkK9ks5qaFz1gaJpZM4JW8NK>
> .
>